src/ray/raylet/node_manager.cc

```diff
@@ -2440,6 +2442,48 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &node_stats_
   }
 }
 
+void NodeManager::HandleGetResourceUsageByTask(
+    const rpc::GetResourceUsageByTaskRequest &request,
+    rpc::GetResourceUsageByTaskReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(INFO) << "HandleGetResourceUsageByTask: lease workers: "
+                << leased_workers_.size();
+  auto total_map = reply->mutable_total();
+  auto available_map = reply->mutable_available();
+  auto node_resources =
+      cluster_resource_scheduler_->GetLocalResourceManager().GetLocalResources();
+  for (const auto &resource : node_resources.total.ToResourceRequest().ToResourceMap()) {
+    (*total_map)[resource.first] = resource.second;
+  }
+  for (const auto &resource :
+       node_resources.available.ToResourceRequest().ToResourceMap()) {
+    (*available_map)[resource.first] = resource.second;
+  }
+  // We get the logical resources acquired by each worker that the raylet has leased out
+  RAY_LOG(INFO) << "HandleGetResourceUsageByTask: lease workers: "
+                << leased_workers_.size();
+  for (const auto &worker_it : worker_pool_.GetAllRegisteredWorkers(
```

Comment:
Can you just create a new API instead of adding `filter_inactive_workers `? It is a bit weird because for example, the semantic of filter_dead_workers=false && filter_inactive_workers=true is not clear (this will still filter dead workers). Maybe we can create `worekr_pool_.GetWorkersInUse` and then in the docstring, we can clearly define what "in use" mean (meaning worker is doing some work using ray resources).